src/Components/Calendar/Calendar.tsx

```diff
@@ -0,0 +1,277 @@
+import type { Story } from "@ladle/react";
+import "./big-calendar.css";
+import styles from "./Calender.module.css";
+import { Calendar as BigCalendar, momentLocalizer } from "react-big-calendar";
+import Toolbar from "react-big-calendar/lib/Toolbar";
+import { Views } from "react-big-calendar";
+import moment from "moment";
+import { useCallback, useState } from "react";
+import { Toggle } from "../Toggle/ToggleSwitch/Toggle";
+import { EventPopup } from "./EventPopup/EventPopup";
+import { convertToDateTimeLocalString } from "../../Utils/convertToDateTimeLocalString";
+import {
+  CalendarEvent,
+  PopupEvent,
+  defaultEventLength as defaultEveLength,
+  defaultView,
+} from "./calendarutils/calendarutils";
+
+const localizer = momentLocalizer(moment);
+
+type Keys = keyof typeof Views;
+
+type Props = {
+  events: CalendarEvent[];
+  backgroundEvents: { id: number; title: string; start: Date; end: Date };
+  availability?: { start: string; end: string };
+  defaultEventLength?: number;
+  athletes: [{ option: string; value: string }];
+};
+
+export const Calendar: Story<Props> = ({
+  events,
+  backgroundEvents,
+  availability,
+  defaultEventLength = defaultEveLength,
+  athletes,
+}: Props) => {
+  const [date, setDate] = useState<Date>(moment(new Date()).toDate());
+  const [view, setView] = useState<(typeof Views)[Keys]>(Views.MONTH);
+  const [toggleValue, setToggleValue] = useState(defaultView);
+  const [myEvents, setEvents] = useState<CalendarEvent[]>(events);
+  const [selected, setSelected] = useState();
+  const [eventPopup, setEventPopup] = useState<PopupEvent>();
+  const [show, setShow] = useState(false);
+
+  const currentMonth = date.toLocaleString("default", { month: "long" });
+
+  const toolbarTitle = new Map();
+  toolbarTitle.set("month", `${currentMonth} ${date.getFullYear()}`);
+  toolbarTitle.set("week", `${currentMonth} ${date.getFullYear()}`);
+  toolbarTitle.set(
+    "day",
+    `${date.getDate()}-${date.getMonth() + 1}-${date.getFullYear()}`
+  );
+
+  class CustomToolbar extends Toolbar {
```

Comment:
Might be better to split components out into their own file within this directory